whitesource/util.py

```diff
@@ -1,10 +1,167 @@
+
+    ttable_header = ('Component', 'Greatest CVSS-V3', 'Corresponding CVE')
+    ttables = []
+
+    for source in above, below:
+        if len(source) == 0:
+            ttables.append("")
+            continue
+        ttable_data = (
+            (
+                component,
+                dic["CVSS-V3"],
+                dic["CVE"]
+            ) for component, dic, in source
+        )
+
+        ttable = tabulate.tabulate(
+            headers=ttable_header,
+            tabular_data=ttable_data,
+            tablefmt=tablefmt,
+            colalign=('left', 'center', 'center')
+        )
+
+        ttables.append(ttable)
+
+    return ttables
+
+
+def assemble_mail_body(tables: typing.List,
+                       threshold: float):
+    return f'''
+        <div>
+            <p>
+                Note: you receive this E-Mail, because you were configured as a mail recipient
+                (see .ci/pipeline_definitions)
+                To remove yourself, search for your e-mail address in said file and remove it.
+            </p>
+            <br></br>
+            <p>
+                The following component(s) have a CVSS-V3 greater than the configured threshold of
+                {threshold}. It is configured at the
+                <a href="https://github.wdf.sap.corp/kubernetes/cc-config">
+                    pipeline definition
+                </a>.
+            </p>
+            {tables[0]}
+            <br></br>
+            <br></br>
+            <p>
+                These are the remaining component(s) with a CVSS-V3 lower than {threshold}
+            </p>
+            {tables[1]}
+            <br></br>
+            <br></br>
+            <p>
+                WhiteSource triage has to be done on the
+                <a href="https://saas.whitesourcesoftware.com/Wss/WSS.html#!alertsReport">
+                    WhiteSource Alert Reporting
+                </a>
+                page. Appropriate filters have to applied manually, "Gardener" is a matching keyword.
+            </p>
+        </div>
+    '''
+
+
+def find_greatest_cve(projects,
+                      client):
+    report = {}
+
+    # get all projects for product
+    for project in projects["projects"]:
+        pname = project["projectName"]
+        ptoken = project["projectToken"]
+
+        # get vulnerability report per project
+        ci.util.info(f'retrieving project vulnerability report for {pname}')
+        pvr = client.get_project_vulnerability_report(project_token=ptoken)
+
+        # find greatest cve per project
+        for vul in pvr["vulnerabilities"]:
+            try:
+                report = _add_cve_entry(report=report,
+                                        cvss_key_name="cvss3_score",
+                                        pname=pname,
+                                        vul=vul)
+            except KeyError:
+                # https://github.com/gardener/cc-utils/pull/476#discussion_r490231239
```

Comment:
it might be nice to add also the quoted link and part of the response to this piece of code, as it - as discussed - is not obvious (I think)